src/rust/analyzer-dispatcher/src/main.rs

```diff
@@ -135,15 +134,21 @@ struct AnalyzerDispatcher {
 impl AnalyzerDispatcher {
     pub async fn new(
         config: AnalyzerDispatcherConfig,
-        plugin_work_queue_client: PluginWorkQueueServiceClient,
+        plugin_work_queue_client: PluginWorkQueueClient,
     ) -> Result<Self, ConfigurationError> {
         let graph_update_consumer: Consumer<Update> = Consumer::new(config.kafka_config)?;
         let graph_update_retry_producer: RetryProducer<Update> =
             RetryProducer::new(config.kafka_retry_producer_config)?;
-        let client_config = PluginRegistryClientConfig::parse();
 
-        let plugin_registry_client =
-            PluginRegistryServiceClient::connect_with_config(client_config).await?;
+        let client_config = Figment::new()
+            .merge(Env::prefixed("PLUGIN_REGISTRY_CLIENT_"))
+            .extract()?;
+        let plugin_registry_client = PluginRegistryClient::connect_with_healthcheck(
```

Comment:
Colin very adamantly pushed against our having healthchecks in our code - this is a discussion best hashed out with him. (I'm in favor of the health checks, personally.)

https://grapl-internal.slack.com/archives/C03JNJ57A9G/p1658862938530339

this is from before we introduced client-executor's retries stuff

Reply:
fixed